src/private/MDMRegisteredAnimation.m

```diff
@@ -0,0 +1,39 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDMRegisteredAnimation.h"
+
+@implementation MDMRegisteredAnimation
+
+- (instancetype)initWithKey:(NSString *)key animation:(CABasicAnimation *)animation {
+  self = [super init];
+  if (self) {
+    _key = [key copy];
+    _animation = animation;
+  }
+  return self;
+}
+
+- (NSUInteger)hash {
+  return _key.hash;
+}
+
+- (BOOL)isEqual:(id)object {
+  return [_key isEqual:object];
```

Comment:
Two registered animations are equal if they have the same key but different animations?  Could you please add some basic unit tests for this class?

Reply:
We don't have a convention in place for private API unit tests unfortunately, meaning we can't access the APIs from unit test targets without some change to both our podfile & bazel configs.